src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -74,23 +74,50 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 	fun getTracks() =
 		parseJSON(TrackResponse::class.java)?.results
 	
+	fun postLogin(username: String, password: String) =
+		post(HttpPost(uri).apply {
+			setHeader("Accept", "application/json")
+			setHeader("Content-type", "application/json")
+			entity = StringEntity("{\"email\":\"$username\",\"password\":\"$password\"}")
+		})
+	
+	fun postLogout() =
+		post(HttpPost(uri).apply {
+			setHeader("Accept", "application/json")
+			setHeader("Content-type", "application/json")
+		})
+	
 	/** Aborts this connection and thus terminates the InputStream if active */
 	fun abort() {
 		httpGet?.abort()
+		httpPost?.abort()
+		httpPut?.abort()
 	}
 	
 	// Direct Requesting
 	
 	private var httpGet: HttpGet? = null
-	fun execute() {
+	fun get() {
 		httpGet = HttpGet(uri)
 		response = execute(httpGet!!)
 	}
 	
+	private var httpPost: HttpPost? = null
+	fun post(request : HttpPost) {
+		httpPost = request
+		response = execute(httpPost!!)
```

Comment:
```suggestion
		response = execute(request)
```
that way you don't have to use `!!`, these should be avoided if possible